build/targets/standard-lifecycle.targets

```diff
@@ -125,7 +159,51 @@ that matches "$(RepositoryRoot)/shared/*.Sources".
     <MSBuild Targets="Pack"
       Projects="$(MSBuildThisFileDirectory)..\shared\sharedsources.csproj"
       Properties="PackageOutputPath=$(BuildDir);RepositoryRoot=$(RepositoryRoot);NuspecBasePath=%(SharedSourceDirectories.Identity);PackageId=%(FileName)%(Extension)"
-      Condition="'@(SharedSourceDirectories)'!=''" />
+      Condition="'@(SharedSourceDirectories)'!=''"
+      BuildInParallel="$(BuildInParallel)" />
+  </Target>
+
+<!--
+###################################################################
+Target: TestProjects
+
+Items: ProjectsToTest. Defaults to test/*/*.csproj
+Properties: ExcludeFromTest. ItemSpec for projects to avoid testing.
+
+Runs the VSTest on all projects in the ProjectsToTest itemgroup.
+###################################################################
+-->
+
+  <PropertyGroup>
+    <!-- TODO if VS Test doesn't make minimal output the default log setting, we can set that here. cref https://github.com/Microsoft/vstest/issues/301 -->
+    <VSTestLogger Condition=" '$(VSTestLogger)'=='' AND '$(TEAMCITY_VERSION)' != '' ">trx</VSTestLogger>
+    <IgnoreFailingTestProjects>false</IgnoreFailingTestProjects>
+    <ContinueOnTestError Condition="'$(KOREBUILD_IGNORE_DOTNET_TEST_EXIT_CODE)' == '1'">true</ContinueOnTestError>
+  </PropertyGroup>
+
+  <Target Name="TestProjects">
+
+    <ItemGroup Condition="'@(ProjectsToTest)'==''">
+      <!-- put unit test projects ahead of functional tests -->
```

Comment:
Do we actually have this as a convention? `Tests` -> Functional? Otherwise this is kinda misleading.